alerter.c

```diff
@@ -11,21 +13,41 @@ int networkAlertStub(float celcius) {
     return 200;
 }
 
-void alertInCelcius(float farenheit) {
+int networkAlertReal(float celcius) {
+    printf("ALERT: Temperature is %.1f celcius.\n", celcius);
+    // Return 200 for ok
+    // Return 500 for not-ok
+    // Assumption If celcius is less than 200 for ok else not-ok 
+    if (celcius < 200.0) {
+        return 200;
+    }else {
+        return 500;
+    }
+
+}
+
+void alertInCelcius(networkAlert_fnptr networkAlert_fn, float farenheit) {
     float celcius = (farenheit - 32) * 5 / 9;
-    int returnCode = networkAlertStub(celcius);
+    int returnCode = networkAlert_fn(celcius);
     if (returnCode != 200) {
         // non-ok response is not an error! Issues happen in life!
         // let us keep a count of failures to report
         // However, this code doesn't count failures!
         // Add a test below to catch this bug. Alter the stub above, if needed.
-        alertFailureCount += 0;
+        alertFailureCount += 1;
     }
 }
 
 int main() {
-    alertInCelcius(400.5);
-    alertInCelcius(303.6);
+    alertInCelcius(networkAlertStub, 500.5);
+    assert(alertFailureCount == 1);
+    alertInCelcius(networkAlertStub, 600.6);
+    assert(alertFailureCount == 2);
+    alertInCelcius(networkAlertStub, 1000.1);
+    assert(alertFailureCount == 1);

```

Comment:
alertFailureCount looks like a counter. Is it possible that the failure count reduce by 1?